pkg/ccl/kvccl/kvtenantccl/upgradeinterlockccl/BUILD.bazel

```diff
@@ -5,17 +5,20 @@ go_test(
     name = "upgradeinterlockccl_test",
     size = "enormous",
     srcs = [
+        "generated_test.go",
```

Comment:
*Consider* having this file not be checked-in and just generate it as part of the Bazel build rule. Then you don't have to update `bazel-generate.sh`, and there is no possibility to forget to do this generation which would cause you to hit a roadblock in CI. `dev gen go` would populate this test file if you need it for a non-Bazel build.

Reply:
Do you mean adding a `genrule` for that?

Reply:
Yes, that is what I mean.

Generally generated code should not be checked in. It's annoying if it is checked in because it's just another thing that can fail CI.

Some generated code is checked in for various reasons. For example for logictests which generation creates many directories, we want that stuff checked in because the packages don't even exist unless that generation has run. For this, I don't know.

Reply:
Thanks for explaining, done.